test/index.js

```diff
@@ -196,12 +196,81 @@ describe('simple-keyring', () => {
   })
 
   describe('#signTypedData', () => {
+    const address = '0x29c76e6ad8f28bb1004902578fb108c507be341b'
+    const privKeyHex = '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0'
+  
+    const privKey = Buffer.from(privKeyHex, 'hex')
+  
+    const typedData = [
+      {
+        type: 'string',
+        name: 'message',
+        value: 'Hi, Alice!'
+      }
+    ]
+    const msgParams = { data: typedData }
+
+    it('returns the expected value', async () => {
+      await keyring.deserialize([privKeyHex])
+      const sig = await keyring.signTypedData(address, typedData)
+      const signedParams = Object.create(msgParams)
+      signedParams.sig = sig;
+      const restored = sigUtil.recoverTypedSignatureLegacy(signedParams)
+      assert.equal(restored, address, 'recovered address')
+    })
+  })
+
+  describe('#signTypedData_v1', () => {
+    const address = '0x29c76e6ad8f28bb1004902578fb108c507be341b'
+    const privKeyHex = '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0'
+  
+    const privKey = Buffer.from(privKeyHex, 'hex')
+  
+    const typedData = [
+      {
+        type: 'string',
+        name: 'message',
+        value: 'Hi, Alice!'
+      }
+    ]
+    const msgParams = { data: typedData }
+
+    it('returns the expected value', async () => {
+      await keyring.deserialize([privKeyHex])
+      const sig = await keyring.signTypedData_v1(address, typedData)
+      const signedParams = Object.create(msgParams)
+      signedParams.sig = sig;
+      const restored = sigUtil.recoverTypedSignatureLegacy(signedParams)
+      assert.equal(restored, address, 'recovered address')
+    })
+  })
+
+  describe('#signTypedData_v3', () => {
     const address = '0x29c76e6ad8f28bb1004902578fb108c507be341b'
     const privKeyHex = '0x4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0'
 
     it('returns the expected value', async () => {
-      const expectedSignature = '0xf2951a651df0a79b29a38215f9669b06499fa45d3b41c7acedd49c1050e8439f3283156a0797113c9c06c1df844495071aaa5721ea39198b46bf462f7417dfba1b'
```

Comment:
We might want to test the actual signatures, to ensure they don't break between versions. Otherwise any signature changes would break any existing signatures in the wild.

Reply:
Re-added this signature enforcement, along with all others from [the test case gist](https://gist.github.com/danfinlay/750ce1e165a75e1c3387ec38cf452b71).

Bizzarely: `v4` test is failing for lack of array encoding support, while the exact same test seems to pass on `eth-sig-util`. PRing here so the error is visible on CI.